iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/VoltageLevelImpl.java

```diff
@@ -618,4 +619,91 @@ public void allocateVariantArrayElement(int[] indexes, int sourceIndex) {
+            if (newTopologyKind == TopologyKind.NODE_BREAKER) {
+                throw new PowsyblException("Topology model conversion from bus/breaker to node/breaker not yet supported");
+            } else if (newTopologyKind == TopologyKind.BUS_BREAKER) {
+                BusBreakerTopologyModel newTopologyModel = new BusBreakerTopologyModel(this);
+
+                for (Bus bus : topologyModel.getBusBreakerView().getBuses()) {
+                    // no notification, this is just a mutation of a calculation bus to a configured bus
+                    ConfiguredBusImpl configuredBus = new ConfiguredBusImpl(bus.getId(), bus.getOptionalName().orElse(null), bus.isFictitious(), this);
+                    newTopologyModel.addBus(configuredBus);
+                }
+
+                // transfer retained switches
+                for (Switch sw : topologyModel.getBusBreakerView().getSwitchStream().toList()) {
+                    String busId1 = topologyModel.getBusBreakerView().getBus1(sw.getId()).getId();
+                    String busId2 = topologyModel.getBusBreakerView().getBus2(sw.getId()).getId();
+                    // no notification, this is just a transfer
+                    ((NodeBreakerTopologyModel) topologyModel).removeSwitchFromTopology(sw.getId(), false);
+                    newTopologyModel.addSwitchToTopology((SwitchImpl) sw, busId1, busId2);
+                }
+
+                // reconnect all connectable to new topology model
+                // first store all bus/breaker topological infos associated to this terminal because we will start moving
+                // terminal from old mode to new one, it will modify the old topology model
+                List<Triple<TerminalExt, Bus, Boolean>> oldTopologyModelInfos = new ArrayList<>();
+                for (Terminal oldTerminal : topologyModel.getTerminals()) {
+                    Bus connectableBus = oldTerminal.getBusBreakerView().getConnectableBus();
+                    boolean connected = oldTerminal.isConnected();
+                    oldTopologyModelInfos.add(Triple.of((TerminalExt) oldTerminal, connectableBus, connected));
+                }
+
+                for (var triple : oldTopologyModelInfos) {
+                    TerminalExt oldTerminalExt = triple.getLeft();
+                    Bus connectableBus = triple.getMiddle();
+                    boolean connected = triple.getRight();
+
+                    // if there is no way to find a connectable bus, remove the connectable
+                    // an alternative would be to connect them all to a new trash configured bus
+                    if (connectableBus != null) {
+                        AbstractConnectable<?> connectable = oldTerminalExt.getConnectable();
+
+                        // create the new terminal with new type
+                        TerminalExt newTerminalExt = null;
+                        if (oldTerminalExt.getConnectable().getType() != IdentifiableType.BUSBAR_SECTION) {
```

Comment:
In case of a BBS, its terminal is set to null but it's still in the connectable index (this can be checked in the unit test), so maybe you could move this condition in the previous if above `if (connectableBus != null) {`

Reply:
There was a bug indead, busbar sections were still reacheable from the network index. I now start the conversion by removing the busbar sections.